archaius2-core/src/main/java/com/netflix/archaius/ConfigProxyFactory.java

```diff
@@ -368,17 +356,26 @@ protected <T> MethodInvoker<T> createInterfaceProperty(String propName, final T
         return (args) -> proxy;
     }
 
-    protected <T> MethodInvoker<T> createScalarProperty(final Type type, final String propName, Supplier<T> next) {
+    protected <T> MethodInvoker<T> createScalarProperty(final Type type, final String propName, Function<Object[], T> next) {
```

Comment:
This is an API change. But I doubt anyone is actually overriding these methods.
I looked internally and couldn't any reference.